code generators/de.fhdo.lemma.model_processing.code_generation.container_base/src/main/java/de/fhdo/lemma/model_processing/code_generation/container_base/code_generation/DockerComposeCodeGenerator.xtend

```diff
@@ -0,0 +1,85 @@
+package de.fhdo.lemma.model_processing.code_generation.container_base.code_generation
+
+import de.fhdo.lemma.model_processing.annotations.CodeGenerationModule
+import de.fhdo.lemma.operation.intermediate.IntermediateOperationModel
+import de.fhdo.lemma.model_processing.builtin_phases.code_generation.AbstractCodeGenerationModule
+import de.fhdo.lemma.operation.intermediate.IntermediatePackage
+import de.fhdo.lemma.operation.intermediate.IntermediateOperationNode
+import de.fhdo.lemma.model_processing.languages.LanguageDescription
+import de.fhdo.lemma.model_processing.code_generation.container_base.file.docker.OpendDockerComposeFile
+import java.util.Map
+import java.util.HashMap
+import java.io.File
+import org.jetbrains.annotations.NotNull
+
+/**
+ * Main class of the docker compose generation module of the container base code generator.
+ *
+ * @author [Philip Wizenty](mailto:philip.wizenty@fh-dortmund.de)
+ */
+@CodeGenerationModule(name="dockercompose")
+class DockerComposeCodeGenerator extends AbstractCodeGenerationModule {
+
+    var IntermediateOperationModel model
+    var Map<String, String> content
+
+    /*
+     * Main method for the creation of the docker-compose file
+     */
+    @NotNull
+    override execute(String[] args, String[] moduleArgs) {
+        // Receive the intermediate operation model
+        model = this.intermediateModelResource.contents.get(0) as IntermediateOperationModel
+
+        // Create a Map which contains the generated source code
+        content = new HashMap
+
+        /*
+         * Set targetFolder for docker-compose.yml file and open an existing or create a new
+         * docker-compose file
+         */
+        OpendDockerComposeFile.instance.init(targetFolder)
+        OpendDockerComposeFile.instance.openExistingDockerComposeFile
+
+        // Add docker-compose parts of the container to the docker compose file
+        model.containers.forEach[container |
+            this.createDockerComposeFile(container)
+        ]
+
+        // Add docker-compose parts of the infrastructure node to the docker compose file
+        model.infrastructureNodes.forEach[node |
+            this.createDockerComposeFile(node)
+        ]
+
+        // CreateDockerComposeFile
+        content.put(targetFolder + File.separator + "docker-compose.yml",
```

Comment:
Put such statements in a variable and make the `put`-call single-lined for the sake of readability.